foedus-core/src/foedus/xct/xct_id.cpp

```diff
@@ -212,5 +324,16 @@ std::ostream& operator<<(std::ostream& o, const LockableXctId& v) {
   return o;
 }
 
+std::ostream& operator<<(std::ostream& o, const McsRwLock& v) {
+  o << "<McsRwLock><locked>" << v.is_locked() << "</locked><tail_waiter>"
+    << v.get_tail_waiter() << "</tail_waiter><tail_block>" << v.get_tail_waiter_block()
```

Comment:
readers_count and next_writer field.
this is debug-write, no worry about performance. so you can just use seq-cst load here.